app/controllers/cinema_halls_controller.rb

```diff
@@ -1,40 +1,38 @@
 # frozen_string_literal: true
 
 class CinemaHallsController < ApplicationController
-  before_action :set_cinema_halls, only: %i[show update]
-  # GET /cinema_halls
+
+
   def index
-    @cinema_halls = CinemaHall.all.map do |cinema_hall|
-      cinema_hall_hash(cinema_hall)
-    end
-    render json: @cinema_halls
+    cinema_halls = CinemaHalls::UseCases::ShowAll.new.call
+    render json: CinemaHalls::Representers::Multiple.new(cinema_halls).call
   end
 
-  # GET /cinema_halls/1
   def show
-    render json: @cinema_hall
+     cinema_hall = CinemaHalls::UseCases::Show.new(id: params[:id]).call
+     render json: CinemaHalls::Representers::Single.new(cinema_hall).call
   end
 
-  # POST /cinema_halls
   def create
-    @cinema_hall = CinemaHall.new(cinema_hall_params)
-
-    if @cinema_hall.save
-      render json: cinema_hall_hash(@cinema_hall), status: :created, location: @cinema_hall
+    cinema_hall = CinemaHalls::UseCases::Create.new(params: cinema_hall_params).call
+    if cinema_hall.valid?
+      render CinemaHalls::Representers::Single.new(cinema_hall).call, status: :created
     else
-      render json: @cinema_hall.errors, status: :unprocessable_entity
+      render json: CinemaHall.errors, status: :unprocessable_entity
     end
   end
 
-  # PATCH/PUT /cinema_halls/1
   def update
-    if @cinema_hall.update(cinema_hall_params)
-      render json: @cinema_hall
+    updated_cinema_hall = CinemaHalls::UseCases::Update.new(params: cinema_hall_params, id: params[:id]).call
+    if updated_cinema_hall.errors.any?
+      render updated_cinema_hall.errors, status: :unprocessable_entity
```

Comment:
To catch errors, use `rescue` keyword. If statements are no great to catch errors. I did the same when I started bootcamp 3 months ago.
https://blog.appsignal.com/2018/04/10/rescuing-exceptions-in-ruby.html
Delete if statement, put rescue [ActiveRecord::RecordNotFound](https://api.rubyonrails.org/classes/ActiveRecord/RecordNotFound.html)